### PR TITLE
Allow registering click and value change events after element instantiation

### PR DESCRIPTION
**Pull request by falkoschindler**
This PR addresses feature request #2687, allowing to add event handlers after UI element have been created. It is still work in progress though.

- [x] button
- [x] value elements
- [x] color_picker
- [x] echart
- [x] interactive_image
- [x] joystick
- [x] json_editor
- [x] keyboard
- [x] list
- [x] menu_item
- [x] scene
- [x] scroll_area
- [x] table
- [x] tree
- [x] upload

### CODE REVIEW

File: nicegui/elements/button.py [hunk 1]
```diff
@@ -35,7 +35,11 @@ def __init__(self,
             self._props['icon'] = icon
 
         if on_click:
-            self.on('click', lambda _: handle_event(on_click, ClickEventArguments(sender=self, client=self.client)), [])
+            self.on_click(on_click)
+
+    def on_click(self, callback: Callable[..., Any]) -> None:
+        """Set a callback to be invoked when the button is clicked."""
+        self.on('click', lambda _: handle_event(callback, ClickEventArguments(sender=self, client=self.client)), [])
```
Comment (petergaultney) [inline nicegui/elements/button.py:42]: i think i'd appreciate if this were the actual builder pattern, and we did

```suggestion
    def on_click(self, callback: Callable[..., Any]) -> Self:
        """Set a callback to be invoked when the button is clicked."""
        self.on('click', lambda _: handle_event(callback, ClickEventArguments(sender=self, client=self.client)), [])
        return self
```

Comment (falkoschindler) [reply nicegui/elements/button.py:42]: Oh sure, I forgot to return `self`. 👍

**Comment by petergaultney** [comment]
this is totally unrelated to this PR (but very related to my attempts to contribute):

i just ran `autopep8 --max-line-length=120 --experimental  --in-place --recursive .` (as per the README) on the directory after making my changes, and a whole bunch of files that I hadn't touched got reformatted. That seems... undesirable. Is this a known issue?

Screenshot 2024-03-15 at 5 17 04 PM

**Comment by falkoschindler** [comment]
Regarding the autopep8 conflict, I just created a separate issue: https://github.com/zauberzeug/nicegui/issues/2750.

**Review by rodja — COMMENTED** [review]
Should we add some tests?

**Pull request merged**